### DISCUSSION TITLE
how to catch auth network request error?

### DISCUSSION THREAD

**Discussion by daveteu** [invertase/react-native-firebase#4356] [Q&A]
Hi,

How can I catch the network error generated by onAuthStateChange?

I have a listener code below, I wish to catch the error generated in react native but it's still generating the warning.
```
 const unregisterAuthObserver = auth().onAuthStateChanged(async (currentUser, error) => {

            if (error) return console.log('[ FIREBASE AUTHENTICATION ERROR', error); <-- this doesn't work

})
```

<img width="408" alt="image" src="https://user-images.githubusercontent.com/18649392/95080424-ac9ee180-074a-11eb-910d-6e25aa7add3e.png">

**Comment by mikehardy** [comment]
do you have evidence that the method was called? That's for state change errors but I am guessing that whatever auth method you called that could have resulted in a state change caused the error.

Whatever method call that was should be in a try catch.

I receive these errors sometimes but they are always in a try/catch and I log them with the app responding appropriately

**Reply by daveteu** [reply]
hmm no I guess it may not be at authstatechanged but when it tries to refresh token.

The test I'm doing now is offline handling, it appears if I turn off internet, open the app, and I will see this error. Could be firebase trying to refresh token or something? I do not know where to capture that error then.

**Reply by mikehardy** [reply]
If I understand javascript correctly, you can't have an unhandled promise rejection without calling an async function outside a try/catch. Showing an image vs showing the text of the full stack trace is not as instructive as it could be, there will be a full stack trace somewhere and it will point to the problem.
